Add readme for new palette service (#84512)
elastic/kibana: src/plugins/charts/public/services/palettes/README.md

# Palette Service

> The `palette` service offers a collection of palettes which can be used to consistently assign colors to various parts of charts. All palettes implement a uniform interface, allowing a consumer to implement it in a generic way and exposing a palette picker to the user to switch palettes

Reviewer: Trying to make this more concise, but taking this suggestion is optional

```suggestion
The `palette` service offers a collection of palettes which implement a uniform interface for assigning colors to charts. The service provides methods for switching palettes
```

easily. It's used by the x-pack plugins `canvas` and `lens`.

Each palette is allowed to store some state as well which has to be handled by the consumer.

Palettes are integrated with the expression as well using the `system_palette` and `palette` functions.

## Using the palette service

> To consume the palette service, use `charts.palettes.getPalettes` to lazily load the async bundle implementing existing palettes. This should not happen as part of the `setup` or `start` phases of a consuming plugin, but only if it's certain palettes will actually be required.

Reviewer: More edits for conciseness

```suggestion
To consume the palette service, use `charts.palettes.getPalettes` to lazily load the async bundle implementing existing palettes. This is recommended to be called in the renderer, not as part of the `setup` or `start` phases of a plugin.
```
